engineyard/ey-cookbooks-dev-v7
# [FBZ-10451] Fail2ban Support

Description of your patch
-------------
Adds support to install `fail2ban` on Engineyard `stack-v7`

Recommended Release Notes
-------------
Provides `fail2ban` support for EYC stack-v7. Via this fail2ban could be configured via Environment Variables on Instances. Moreover, fail2ban could be customized to be installed on specific Instance roles.

Estimated risk
-------------
Medium

Components involved
-------------
`ey-fail2ban` and `ey-base`

Dependencies
-------------
`ey-base`

Description of testing done
-------------
Validated on a solo environment.

1. Set `EY_FAIL2BAN_ENABLED` to true via `environment-variables` on dashboard.
2. Apply recipes for the environment
3. Check `service fail2ban status` to see if fail2ban is running

QA Instructions
-------------

* On Solo environment
1. Set `EY_FAIL2BAN_ENABLED` to true via `environment-variables` on dashboard.
2. Apply recipes for the environment
3. Check `service fail2ban status` to see if fail2ban is running

* On Multi-environment
1. Set `EY_FAIL2BAN_ENABLED` to true via `environment-variables` on dashboard.
2. Apply recipes for the environment
3. Check `service fail2ban status` to see if fail2ban is running on each of the instances.
4. Check `/etc/fail2ban/jail.local` to see if nginx jails are not reflected on database and util instances

* On Multi-environment only in app-instances
1. Set `EY_FAIL2BAN_ENABLED` to true and `EY_FAIL2BAN_INSTANCE_ROLES` to `app,app_master` (comma,separated) via `environment-variables` on dashboard.
2. Apply recipes for the environment
3. Check `service fail2ban status` to see if fail2ban is running on each of the instances.
4. Check `/etc/fail2ban/jail.local` to see if nginx jails are not reflected on database and util instances
